src/Extend/Notification.php

```diff
@@ -38,27 +39,38 @@ public function type(string $blueprint, string $serializer, array $channelsEnabl
      * @param string $driverName The name of the notification driver.
      * @param string $driver The ::class attribute of the driver class.
      *                       This driver should implement \Flarum\Notification\Driver\NotificationDriverInterface.
+     * @param string[] $typesEnabledByDefault The names of blueprint classes of types enabled by default for this driver.
      * @return self
      */
-    public function driver(string $driverName, string $driver)
+    public function driver(string $driverName, string $driver, array $typesEnabledByDefault = [])
     {
-        $this->drivers[$driverName] = $driver;
+        $this->drivers[$driverName] = [$driver, $typesEnabledByDefault];
 
         return $this;
     }
 
     public function extend(Container $container, Extension $extension = null)
     {
         $container->extend('flarum.notification.blueprints', function ($existingBlueprints) {
-            return array_merge($existingBlueprints, $this->blueprints);
+            $existingBlueprints = array_merge($existingBlueprints, $this->blueprints);
+
+            foreach ($this->drivers as $driverName => [$driver, $typesEnabledByDefault]) {
+                foreach ($typesEnabledByDefault as $blueprintClass) {
+                    if (isset($existingBlueprints[$blueprintClass]) && (! in_array($driverName, $existingBlueprints[$blueprintClass]))) {
+                        $existingBlueprints[$blueprintClass][] = $driverName;
+                    }
+                }
+            }
+
+            return $existingBlueprints;
         });
 
         $container->extend('flarum.api.notification_serializers', function ($existingSerializers) {
             return array_merge($existingSerializers, $this->serializers);
         });
 
         $container->extend('flarum.notification.drivers', function ($existingDrivers) {
-            return array_merge($existingDrivers, $this->drivers);
+            return array_merge($existingDrivers, $this->drivers[0]);
```

Comment:
Isn't `$this->drivers[0]` just the first driver? I think you need `Arr::pluck` here

Reply:
kinda late there, I already pushed a fix :laughing: but I didn't think of using `Arr::pluck`

Reply:
In that case, lets use array_map to avoid unnecessary mutations

Reply:
`array_map` will not do, since it doesn't pass the array keys, only the values, we need the keys. ($driverName => $driver).

we could use a simple foreach for better readability. Or just add another private array property to save the default enabled types for drivers and use that.

Reply:
Urgh I keep forgetting that... Maybe it'd be easier then to just save 2 different arrays, one for the drivers, another for the types.

Reply:
The amount of times I googled if array_map passed keys is insane, I've been doing it constantly for years lmao.

> Maybe it'd be easier then to just save 2 different arrays, one for the drivers, another for the types.

Yea let's do that then.